deepchem/utils/rdkit_util.py

```diff
@@ -0,0 +1,117 @@
+import logging
+
+import os
+import numpy as np
+import tempfile
+import shutil
+from rdkit import Chem
+from rdkit.Chem import AllChem
+from pdbfixer import PDBFixer
+from simtk.openmm.app import PDBFile
+
+
+class MoleculeLoadException(Exception):
+
+  def __init__(self, *args, **kwargs):
+    Exception.__init__(*args, **kwargs)
+
+
+def get_xyz_from_mol(mol):
+  """
+  returns an m x 3 np array of 3d coords
+  of given rdkit molecule
+  """
+  xyz = np.zeros((mol.GetNumAtoms(), 3))
+  conf = mol.GetConformer()
+  for i in range(conf.GetNumAtoms()):
+    position = conf.GetAtomPosition(i)
+    xyz[i, 0] = position.x
+    xyz[i, 1] = position.y
+    xyz[i, 2] = position.z
+  return (xyz)
+
+
+def add_hydrogens_to_mol(mol):
+  molecule_file = None
+  try:
+    molecule_file = str(tempfile.NamedTemporaryFile().name)
+    Chem.MolToPDBFile(mol, molecule_file)
+    fixer = PDBFixer(filename=molecule_file)
+    fixer.addMissingHydrogens(7.4)
+    PDBFile.writeFile(fixer.topology, fixer.positions, open(molecule_file, 'w'))
+    return Chem.MolFromPDBFile(
+        str(molecule_file), sanitize=False, removeHs=False)
+  except ValueError as e:
+    logging.warning("Unable to add hydrogens", e)
+    raise MoleculeLoadException(e)
+  finally:
+    try:
+      os.remove(molecule_file)
+    except (OSError, TypeError):
+      pass
+
+
+def compute_charges(mol):
+  try:
+    AllChem.ComputeGasteigerCharges(mol)
+  except Exception as e:
+    logging.exception("Unable to compute charges for mol")
+    raise MoleculeLoadException(e)
+  return mol
+
+
+def load_molecule(molecule_file, add_hydrogens=True, calc_charges=True):
+  """Converts molecule file to (xyz-coords, obmol object)
+
+  Given molecule_file, returns a tuple of xyz coords of molecule
+  and an rdkit object representing that molecule
+  """
+  if ".mol2" in molecule_file or ".sdf" in molecule_file:
+    suppl = Chem.SDMolSupplier(str(molecule_file), sanitize=False)
+    my_mol = suppl[0]
+  elif ".pdbqt" in molecule_file:
+    raise MoleculeLoadException("Don't support pdbqt files yet")
+  elif ".pdb" in molecule_file:
+    my_mol = Chem.MolFromPDBFile(
+        str(molecule_file), sanitize=False, removeHs=False)
+  else:
+    raise ValueError("Unrecognized file type")
+
+  if my_mol is None:
+    raise ValueError("Unable to read non None Molecule Object")
+
+  if calc_charges:
+    my_mol = add_hydrogens_to_mol(my_mol)
+    compute_charges(my_mol)
+  elif add_hydrogens:
+    my_mol = add_hydrogens_to_mol(my_mol)
+    compute_charges(my_mol)
+
```

Comment:
PS if you're going to be running a fingerprinter on this you probably most definitely want to sanitize this, since the finger prints generated are likely going to depend on perceived properties like Aromaticity, etc.